spk/firebird/src/service-setup.sh

```diff
@@ -0,0 +1,105 @@
+# Fbguard binary is a wrapper around firebird that allows it to run as
+# a daemon
+FBGUARD_BIN_FILE="${SYNOPKG_PKGDEST}/bin/fbguard"
+FB_SECDB="${SYNOPKG_PKGVAR}/security4.fdb"
+
+# We set onetime because we want the systemd to take care about the failures
+SERVICE_COMMAND="${FBGUARD_BIN_FILE} -pidfile ${PID_FILE} -daemon -onetime"
+
+SYSDBA_PASSWORD_FILE="${SYNOPKG_PKGVAR}/SYSDBA.password"
+
+validate_preupgrade () {
+    # We must check that either SYSDBA.password or the security database
+    # exists.
+    # If neither of them is present, we don't know how to set the main
+    # DB password.
+    [ -f "$SYSDBA_PASSWORD_FILE" ] || [ -f "$FB_SECDB" ] || {
+        echo "Neither SYSDBA.password nor security4.fdb exists."
+        exit 1;    
+    };
+}
+
+gen_password() {
+    head -c20 /dev/urandom | base64
+}
+
+service_postinst () {
+    cd "$SYNOPKG_PKGDEST"
+    SET_PASSWD=false
+    SECDB_CREATED=false
+
+    # =================== Build Messages ========================
+    # Build the firebird.msg file during both install and upgrade.
+    # The messages differ across versions and must always match the binaries
+    # version.
+    # Building the messages should be first step because the tools used here
+    # later are already expectine the message file to exist.
+    bin/gbak -rep msg.gbak msg.fdb
+    bin/build_file
+
+    # Publish the message file so that other users can use isql
+    # (isql and other cli tools access the file to translate sql errors)
+    chmod go+r firebird.msg
+
+    echo "Remove unnecessary intermediate files"
+    rm -f bin/build_file msg.gbak msg.fdb
+
+    # =================== Build Security Database ===============
+    # Build security database only when none is present yet or we are being
+    # installed
+    if [ ! -f "$FB_SECDB" ] || [ "$SYNOPKG_PKG_STATUS" = "INSTALL" ];then
+        bin/gbak -rep security4.gbak "$FB_SECDB"
+
+        # We just built the security database. We must also set the password.
+        SECDB_CREATED=true
```

Comment:
With this solution the `else #UPGRADING` block blow (see comment) is fixed can be removed.

```suggestion
        # We just built the security database. We must also set the password.
        SECDB_CREATED=true
        SET_PASSWD=true
```